File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    std::unordered_map<int, int> compliments;
    vector<int> twoSum(vector<int>& nums, int target) {
        auto idx = 0;
        for (const auto& num : nums) {
            auto compliment = target - num;
            if (compliments.contains(compliment)) {
                auto& compliments_idx = compliments[compliment];
                return {idx, compliments_idx};
            }
            compliments.insert(std::make_pair(num, idx));
            idx++;
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map: HashMap<i32, i32> = HashMap::new();
        for(idx, &num) in nums.iter().enumerate() {
            let compliment = target - num;
            if let Some(&comp_idx) = map.get(&compliment) {
                return vec![comp_idx, idx as i32];
            }
            map.insert(num, idx as i32);
        }
        return vec![]
    }
}